src/main/java/DukeManager.java

```diff
@@ -1,35 +1,145 @@
+    private static final Path filePath = Paths.get(".","data", "duke.txt");
+    private static ArrayList<Task> inputs = new ArrayList<>();
 
     /**
      * Returns a string with formatted user inputs.
      *
-     * @param inputs list of user input.
      * @return Formatted string to print
      */
-    public static String formatList(Task[] inputs) {
+
+    public static String formatList() {
         String formattedString = HORIZONTAL_LINE;
-        for (int i = 0; i < inputs.length; i++) {
-            if (inputs[i] == null) {
+        for (int i = 0; i < inputs.size(); i++) {
+            if (inputs.get(i) == null) {
                 break;
             }
-            formattedString += (i+1) + "." + inputs[i].toString() + "\n";
+            formattedString += (i+1) + "." + inputs.get(i).toString() + "\n";
         }
         formattedString += HORIZONTAL_LINE;
         return formattedString;
     }
 
+    public static String saveList() {
+        String formattedString = "";
+        for (int i = 0; i < inputs.size(); i++) {
+            if (inputs.get(i) == null) {
+                break;
+            }
+            formattedString += inputs.get(i).toSaveString() + "\n";
+        }
+        return formattedString;
+    }
+
+    public static void checkExists() throws IOException {
+        boolean directoryExists = Files.exists(directoryPath);
+        boolean fileExists = Files.exists(filePath);
+        if (!directoryExists) {
+            Files.createDirectories(directoryPath);
+        }
+        if (!fileExists) {
+            Files.createFile(filePath);
+        }
+    }
+
+    public static void save() {
+        try {
+            checkExists();
+        } catch (IOException e) {
+            print("Something went wrong when trying to save: " + e.getMessage());
+            return;
+        }
+        String pathName = "./data/duke.txt";
+        try {
+            FileWriter fw = new FileWriter(pathName);
+            fw.write(saveList());
+            fw.close();
+        } catch (IOException e) {
+            print("☹ OOPS!!! Something went wrong when trying to save: " + e.getMessage());
+            return;
+        }
+    }
+
+    public static void load() {
+        if (Files.exists(directoryPath) && Files.exists(filePath)) {
+            File f = new File(String.valueOf(filePath)); // create a File for the given file path
+            Scanner s = null;
+            try {
+                s = new Scanner(f); // create a Scanner using the File as the source
+            } catch (FileNotFoundException e) {
+                print("☹ OOPS!!! Something went wrong when trying to load: " + e.getMessage());
+            }
+            while (s.hasNext()) {
+                String[] line = s.nextLine().split("\\|");
+                Task newTask = null;
+                if (line[0].equals("T")) {
+                    try {
+                        newTask = new Todo(line[2]);
+                    } catch (DukeException e) {
```

Comment:
Some repetitive code here you can consider restructuring?

https://nus-cs2113-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#minimise-code-duplication